clinvar_ingest/parse.py

```diff
@@ -0,0 +1,93 @@
+import gzip
+import json
+import logging
+
+from clinvar_ingest.cloud.gcs import blob_reader, blob_writer
+from clinvar_ingest.fs import BinaryOpenMode
+from clinvar_ingest.fs import _open as _fs_open
+from clinvar_ingest.model import dictify
+from clinvar_ingest.reader import get_clinvar_xml_releaseinfo, read_clinvar_xml
+
+_logger = logging.getLogger("clinvar-ingest")
+
+
+def _open(filepath: str, mode: BinaryOpenMode = BinaryOpenMode.READ):
+    _logger.debug(f"Opening file: {filepath}, mode: {mode}")
+    if filepath.startswith("gs://"):
```

Comment:
Are we able to manage this switch by checking the pydantic typing?  Like... if it's a `GCSBlobPath` or not?

Reply:
This is used in some places where the path is just constructed as a string. Maybe we could refactor those to use a FilePath or GCSBlobPath instead of a raw string.